src/unix/mod.rs

```diff
@@ -694,6 +708,19 @@ extern {
                        res: *mut *mut addrinfo) -> ::c_int;
     pub fn freeaddrinfo(res: *mut addrinfo);
     pub fn gai_strerror(errcode: ::c_int) -> *const ::c_char;
+    #[cfg_attr(all(unix,
+                   not(target_os = "macos"),
+                   not(target_os = "ios"),
+                   not(target_os = "netbsd"),
+                   not(target_os = "openbsd"),
+                   not(target_os = "bitrig"),
+                   not(target_os = "solaris"),
+                   not(target_env = "musl")
```

Comment:
Perhaps this could be converted to a whitelist? It's mostly just targeting Linux/android/FreebSD it looks like?